### ISSUE TITLE
Telegram website, errors in modal dialog does not allow NVDA to change to focus mode or to close the dialog.

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#9230]
### Steps to reproduce:
1. Open
https://web.telegram.org/
2. Log in to telegram
3. Open the submenu right at the top of the webpage and open the settings modal dialog
4. Navigate in browse mode to the link "close" and press enter
5. Press nvda+space to enable focus mode
6. Press escape

### Actual behavior:
When pressing enter on "close", the dialog does not close. When pressing NVDA+spacebar, NVDA does not change to focus mode and you cannot navigate through the dialog anymore. Pressing multiple times escape finally kills the interceptor and closes the dialog. 

### Expected behavior:
NVDA should actually close the dialog when pressing enter on "close". Focus mode should work and escape should also close the dialog.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
Alpha 16691,10950C87

#### Windows version:
Windows 10 1809 update

#### Name and version of other software in use when reproducing the issue:
Firefox 65

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA?
not yet

**Comment by Adriani90** [comment]
In the log file I get following errors:

```
IO - speech.speak (17:14:32.915):
Speaking [u'Link', u'Einstellungen']
IO - inputCore.InputManager.executeGesture (17:14:33.608):
Input: kb(laptop):enter
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:33.730):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:33.737):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
DEBUG - treeInterceptorHandler.update (17:14:33.822):
Adding new treeInterceptor to runningTable: <virtualBuffers.gecko_ia2.Gecko_ia2 object at 0x051DD470>
DEBUG - virtualBuffers.VirtualBuffer._loadBuffer (17:14:33.858):
Buffer load took 0.034 sec, 383 chars
IO - speech.speak (17:14:33.880):
Speaking [u'Dialogfeld']
IO - speech.speak (17:14:33.881):
Speaking [u'Leer']
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:34.042):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:34.049):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:34.058):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
IO - inputCore.InputManager.executeGesture (17:14:34.855):
Input: kb(laptop):downArrow
IO - speech.speak (17:14:34.917):
Speaking [u'Link', u'Schlie\xdfen']
IO - inputCore.InputManager.executeGesture (17:14:35.776):
Input: kb(laptop):downArrow
IO - speech.speak (17:14:35.835):
Speaking [u'Link', u'Bearbeiten']
IO - inputCore.InputManager.executeGesture (17:14:37.677):
Input: kb(laptop):upArrow
IO - speech.speak (17:14:37.730):
Speaking [u'Link', u'Schlie\xdfen']
IO - inputCore.InputManager.executeGesture (17:14:38.148):
Input: kb(laptop):enter
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:38.207):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:38.223):
accRole failed: (-2147024809, 'Falscher Parameter.', (None, None, None, 0, None))
IO - inputCore.InputManager.executeGesture (17:14:39.200):
Input: kb(laptop):NVDA+space
ERROR - scriptHandler.executeScript (17:14:39.210):
error executing script: <bound method GlobalCommands.script_toggleVirtualBufferPassThrough of <globalCommands.GlobalCommands object at 0x04B02C30>> with gesture u'NVDA+Leertaste'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 187, in executeScript
  File "globalCommands.pyc", line 1261, in script_toggleVirtualBufferPassThrough
  File "treeInterceptorHandler.pyc", line 119, in _set_passThrough
  File "review.pyc", line 124, in setCurrentMode
  File "review.pyc", line 26, in getObjectPosition
AttributeError: 'NoneType' object has no attribute 'makeTextInfo'
IO - inputCore.InputManager.executeGesture (17:14:42.555):
Input: kb(laptop):NVDA+f1
ERROR - scriptHandler.executeScript (17:14:42.572):
error executing script: <bound method GlobalCommands.script_navigatorObject_devInfo of <globalCommands.GlobalCommands object at 0x04B02C30>> with gesture 'NVDA+f1'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 187, in executeScript
  File "globalCommands.pyc", line 1467, in script_navigatorObject_devInfo
AttributeError: 'NoneType' object has no attribute 'devInfo'
IO - inputCore.InputManager.executeGesture (17:14:43.811):
Input: kb(laptop):escape
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:43.839):
accRole failed: (-2147417848, 'Das aufgerufene Objekt wurde von den Clients getrennt.', (None, None, None, 0, None))
ERROR - scriptHandler.executeScript (17:14:43.842):
error executing script: <bound method Gecko_ia2.script_disablePassThrough of <virtualBuffers.gecko_ia2.Gecko_ia2 object at 0x051DD470>> with gesture u'Escape'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 187, in executeScript
  File "browseMode.pyc", line 492, in script_disablePassThrough
  File "treeInterceptorHandler.pyc", line 124, in _set_passThrough
AttributeError: 'NoneType' object has no attribute 'treeInterceptor'
IO - inputCore.InputManager.executeGesture (17:14:44.378):
Input: kb(laptop):escape
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IA2Attributes (17:14:44.460):
IAccessibleObject.attributes COMError (-2147417848, 'Das aufgerufene Objekt wurde von den Clients getrennt.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (17:14:44.464):
accRole failed: (-2147417848, 'Das aufgerufene Objekt wurde von den Clients getrennt.', (None, None, None, 0, None))
DEBUG - treeInterceptorHandler.killTreeInterceptor (17:14:44.535):
Killed treeInterceptor: <virtualBuffers.gecko_ia2.Gecko_ia2 object at 0x051DD470>
IO - speech.speak (17:14:44.551):
Speaking [u'reduziert  Untermen\xfc  Link']

**Comment by Adriani90** [comment]
In Chrome the issue is not occuring.

**Comment by Adriani90** [comment]
Issue #5454 might be related.

**Comment by Adriani90** [comment]
Tested in NVDA 2018.4.1 is the same. In NVDA 2018.3.2 I cannot test because there seems to be another bug in it which has been solved later on. In 2018.3.2 I cannot expand the submenu at the top of the webpage.

**Comment by Adriani90** [comment]
This is still reproducible in NVDA 2019.1 and Firefox 66.0.2.
cc: @jcsteh, @leonardder, this is somehow important in my view because as of now NVDA cannot switch to focus mode in that dialog at all and the dialog can only be closed by pressing escape multiple times.

**Comment by jcsteh** [comment]
If you press NVDA+tab after pressing enter to close the dialog, does NVDA say "unknown"? If so, this is https://bugzilla.mozilla.org/show_bug.cgi?id=1190882. I actually have a patch for this, but need to polish it and got sidetracked by other work.

**Comment by Adriani90** [comment]
Yes it say unknown when pressing nvda+tab. Thanks Jamie very much for your reply. That sounds great.

**Comment by jcsteh** [comment]
Fixed for Firefox 68 (and now in Nightly).
